### ISSUE TITLE
convergence in hyperparameter optimization

### ISSUE THREAD

**Issue by SoodabehGhaffari** [chemprop/chemprop#908]
Hello,
I am doing hyperparameter tuning using chemprop_hyperopt for a couple of hyperparameteres. How can I make sure the hyperparameter optimization is converged. I picked number of trials 40 for the following hyperparameteres:
learning_rate warmup_epochs batch_size dropout
As you see, I did not select the number of layers in D-MPNN or FFN and other hyperparameteres related to the architecture of the model, because this is a hyperparameter optimization for a target model initialized/some layers frozen from a source model. I would like to get your opinion on this as well.

Thanks
Best Regards
Soo

**Comment by akshatzalte** [comment]
Hi,
Thanks for asking this question.
Chemprop V2 relies on the raytune package for doing the hyperparameter optimization. A simple check to see if hpopt is converging (or has converged) is to monitor the validation loss in each trial. Ideally, this loss should go down after each trial (this doesn't apply to the initial random points; INFO: `--hyperopt-n-initial-points` is used to specify the number of random initial points).
If the search space is not too large and the validation loss decreases for each trial, finally flattening out, you can be sure that you have reached a minima (at least the local minima).
